.github/workflows/main-checks.yml

```diff
@@ -59,10 +59,10 @@ jobs:
 
       - name: Lint feature soundness
         run: |
-          cargo clippy --release -- --deny=warnings
-          cargo clippy --release --features=ssr -- --deny=warnings
-          cargo clippy --release --features=csr -- --deny=warnings
-          cargo clippy --release --features=hydration -- --deny=warnings
+          cargo clippy --release  --all-targets -- --deny=warnings
+          cargo clippy --release --features=ssr --all-targets -- --deny=warnings
+          cargo clippy --release --features=csr --all-targets -- --deny=warnings
+          cargo clippy --release --features=hydration --all-targets -- --deny=warnings
           cargo clippy --release --features "csr,ssr,hydration,tokio" --all-targets -- --deny=warnings
```

Comment:
Should we just switch to cargo-hack in CI? That will make feature soundness testing easier.

This also needs to be updated in `Makefile.toml` (I should've also done that in #2743, my bad)

Reply:
I tried to switch this to `cargo hack`.

It seems like that cargo-hack uses `cargo check` instead of `cargo clippy`.

We will lose clippy lints if we switch to cargo-hack.